ci/cppcheck/cppcheck-new-issues.sh

```diff
@@ -3,7 +3,7 @@
 scriptdir="$(cd "${0%/*}" && pwd)"
 rootdir="${scriptdir%/*/*}"
 
-# shellcheck source=../../tools/functions.sh
+# shellcheck source=functions.sh
```

Comment:
Although I agree that specifying relative paths is less than ideal, I don't really like the "name-only" approach either.

I know it's not the case right now, but what if at some point we sourced 2 files with the same name, at 2 very different locations in the repository? What would shellcheck do?

Maybe a compromise would be to specify file locations from the top-level folder. This way, we can be sure we always refer to the files we want. 
Invoking shellcheck would be as simple as using `-P "$rootdir"`.
Moving files wouldn't be too hard either, as we can just do a literal search-and-replace to replace `the/old/location` with `the/new/location`.

What do you think?

Reply:
> I know it's not the case right now, but what if at some point we sourced 2 files with the same name, at 2 very different locations in the repository? What would shellcheck do?

I guess that shellcheck behavior may be is undefined in that case. Getting such error may confuse not very familiar with shellcheck developer.

> Invoking shellcheck would be as simple as using -P "$rootdir"

We may strip `-P` at all. Without it, shellcheck looks for scripts pointed by `shellcheck source=` directive starting from folder where scan was started. As result, paths in some script will look like next:
```
# shellcheck source=tools/functions.sh
. "${rootdir}/tools/functions.sh"
```
**Pros**: paths are alike which makes them easy to edit. No need to edit `.gitlab-ci.yml`.
**Cons**: 2 line to change in case `functions.sh` or `owncloud_definitions.sh` are moved.
Which isn't a bad option, overall.

Reply:
Please don't spend too much on this any more - shellcheck has already eaten away way too much of our time...